library/python/CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+cmake_minimum_required(VERSION 3.18)
+project(pywavemap LANGUAGES CXX)
+
+# Warn if the user invokes CMake directly
+if (NOT SKBUILD AND NOT $ENV{CLION_IDE})
+  message(WARNING "\
+  This CMake file is meant to be executed using 'scikit-build-core'.
+  Running it directly will almost certainly not produce the desired
+  result. If you are a user trying to install this package, use the
+  command below, which will install all necessary build dependencies,
+  compile the package in an isolated environment, and then install it.
+  =====================================================================
+   $ pip install .
+  =====================================================================
+  If you are a software developer, and this is your own package, then
+  it is usually much more efficient to install the build dependencies
+  in your environment once and use the following command that avoids
+  a costly creation of a new virtual environment at every compilation:
+  =====================================================================
+   $ pip install nanobind scikit-build-core[pyproject]
+   $ pip install --no-build-isolation -ve .
+  =====================================================================
+  You may optionally add -Ceditable.rebuild=true to auto-rebuild when
+  the package is imported. Otherwise, you need to rerun the above
+  after editing C++ files.")
+endif ()
+
+# Load the wavemap library (if not already loaded)
+if (NOT TARGET wavemap::wavemap_core)
+  if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../cpp/)
+    message(STATUS "Loading wavemap C++ library sources")
+    set(GENERATE_WAVEMAP_INSTALL_RULES OFF)
+    set(BUILD_SHARED_LIBS OFF)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../cpp
+        ${CMAKE_CURRENT_BINARY_DIR}/wavemap)
+  else ()
+    message(ERROR
+        "Can not load wavemap C++ library sources. Are you using an old "
+        "version of pip? If so, retry after upgrading with "
+        "\"python3 -m pip install --upgrade pip\".")
+  endif ()
+endif ()
+
+# Try to import all Python components potentially needed by nanobind
+find_package(Python 3.8
```

Comment:
This requires exactly Python 3.8 correct? Is that really needed as that's quite specific since we're on 3.12 by now already.

Reply:
AFAIK this only requires that a Python version is found that is compatible with Python 3.8. It only requests version 3.8 exactly if we add the `EXACT` keyword. In the tests I ran it compiled without issues on Python 3.12.

Reply:
I read over CMake's doc for the version bit and it's not explicit what they consider "compatible" but by the sound of it and your results it would seem that single version number simply denotes the minimum version required.

Reply:
Yes I think so too - should be safe to leave it at 3.8 :)